pkg/bot/mattermost.go

```diff
@@ -528,6 +523,29 @@ func getBotUserID(ctx context.Context, client *model.Client4, teamID, botName st
 	return teamMember.UserId, nil
 }
 
+func getMattermostTeam(ctx context.Context, client *model.Client4, name string) (*model.Team, error) {
+	botTeams, r, err := client.SearchTeams(ctx, &model.TeamSearch{
+		Term: name, // the search term to match against the name or display name of teams
+	})
+	if err != nil {
+		return nil, fmt.Errorf("while searching team by term: %w", err)
+	}
+
+	if r.StatusCode == http.StatusNotImplemented || len(botTeams) == 0 {
+		// try to check if we can get a given team directly
+		botTeam, _, err := client.GetTeamByName(ctx, name, "")
+		if err != nil {
+			return nil, fmt.Errorf("while getting team by name: %w", err)
+		}
+		botTeams = append(botTeams, botTeam)
+	}
+	if len(botTeams) == 0 {
+		return nil, fmt.Errorf("team %q not found", name)
+	}
+
```

Comment:
Should we handle a case when there are multiple `botTeams`? Maybe return an error that we found multiple (printing theirs name + displayName)?